plugins/kubernetes/app/models/kubernetes/role_validator.rb

```diff
@@ -187,8 +178,9 @@ def validate_containers
       primary_kinds = RoleConfigFile::PRIMARY_KINDS
       containered = templates.select { |t| primary_kinds.include?(t[:kind]) }
       containers = map_attributes([:spec, :containers], elements: containered)
-      return if containers.all? { |c| c.is_a?(Array) && c.size >= 1 }
-      @errors << "#{primary_kinds.join("/")} need at least 1 container"
+      bad = containers.select { |c| !c.is_a?(Array) || c.empty? }
+      return if bad.empty?
+      @errors << "#{containered.map { |r| r[:kind] }.join("/")} need at least 1 container"
```

Comment:
nit: needs not need

Reply:
could be multiple 🤷‍♂️ 
... but I guess will be 1 most of the time ...

Reply:
english is hard. i think it's got to do with adverb vs verb or something but :shrug: